elasticsearch_exporter.go

```diff
@@ -434,82 +195,81 @@ func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
 	resp, err := e.client.Get(e.URI)
 	if err != nil {
 		e.up.Set(0)
-		log.Printf("Error while querying Elasticsearch: %v", err)
+		log.Println("Error while querying Elasticsearch:", err)
 		return
 	}
 	defer resp.Body.Close()
 
 	body, err := ioutil.ReadAll(resp.Body)
-
 	if err != nil {
-		log.Printf("Failed to read ES response body: %v", err)
+		log.Println("Failed to read ES response body:", err)
 		e.up.Set(0)
 		return
 	}
 
 	e.up.Set(1)
 
-	var all_stats NodeStatsResponse
-	err = json.Unmarshal(body, &all_stats)
-
+	var allStats NodeStatsResponse
+	err = json.Unmarshal(body, &allStats)
 	if err != nil {
-		log.Printf("Failed to unmarshal JSON into struct: %v", err)
+		log.Println("Failed to unmarshal JSON into struct:", err)
 		return
 	}
 
-	// Regardless of whether we're querying the local host or the whole
-	// cluster, here we can just iterate through all nodes found.
+	// We only expose metrics for the local node, not the whole cluster.
+	if l := len(allStats.Nodes); l != 1 {
+		log.Println("Unexpected number of nodes returned:", l)
+	}
 
-	for node, stats := range all_stats.Nodes {
-		log.Printf("Processing node %v", node)
+	for _, stats := range allStats.Nodes {
 		// GC Stats
 		for collector, gcstats := range stats.JVM.GC.Collectors {
-			e.counters["jvm_gc_collection_count"].WithLabelValues(all_stats.ClusterName, stats.Name, collector).Set(float64(gcstats.CollectionCount))
-			e.counters["jvm_gc_collection_time_in_millis"].WithLabelValues(all_stats.ClusterName, stats.Name, collector).Set(float64(gcstats.CollectionTime))
+			e.counters["jvm_gc_collections"].WithLabelValues(allStats.ClusterName, collector).Set(float64(gcstats.CollectionCount))
```

Comment:
Maybe this is just my lack of JVM understanding, but why should this metric be "seconds_count"? Just to match the metric with the seconds in it?